### PR TITLE
Create preprod environment

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/00-namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+kind: Namespace
+metadata:
+  name: sentence-planning-preprod
+  labels:
+    cloud-platform.justice.gov.uk/is-production: "false"
+    cloud-platform.justice.gov.uk/environment-name: "preprod"
+  annotations:
+    cloud-platform.justice.gov.uk/business-unit: "HMPPS"
+    cloud-platform.justice.gov.uk/application: "sentence-planning"
+    cloud-platform.justice.gov.uk/owner: "Sentence Planning Team: michael.willis@digital.justice.gov.uk"
```
Comment (digitalronin) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/00-namespace.yaml]: This needs to be a team email address, not an individual.

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: limitrange
+  namespace: sentence-planning-preprod
+spec:
+  limits:
+  - default:
+      cpu: 250m
```
Comment (digitalronin) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml]: Please increase the default cpu limit to 1000m - this is to allow enough capacity for any startup processing to complete quickly.

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      memory: 500Mi
+    defaultRequest:
+      cpu: 125m
```
Comment (digitalronin) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml]: please reduce your cpu default request to 10m. This is not a limit, it's just the amount of CPU the cluster will reserve for each of your containers.

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: namespace-quota
+  namespace: sentence-planning-preprod
+spec:
+  hard:
+    requests.cpu: 3000m
+    requests.memory: 6Gi
+    limits.cpu: 6000m
```
Comment (digitalronin) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml]: Please remove limits.cpu and limits.memory - they have no useful effect.

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: namespace-quota
+  namespace: sentence-planning-preprod
+spec:
+  hard:
+    requests.cpu: 3000m
```
Comment (digitalronin) [inline namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml]: Please reduce requests.cpu for the namespace to 100m

This is not a limit, it's just the amount of CPU the cluster will reserve for your namespace, on creation.

**Comment by digitalronin** [comment]
Also, is this service going to be used for planning *all* sentences? If not, please could you revisit the namespace name, and make sure it's inline with this guidance?

https://ministryofjustice.github.io/technical-guidance/standards/naming-things/#naming-things

**Comment by nickmcmahon01** [comment]
Thanks David, we've taken the name issue back to the team and I'll make the other changes now both to this new environment and our dev one.

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/00-namespace.yaml (hunk 1 above)
Comment (nickmcmahon01) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/00-namespace.yaml]: I fixed this, we have a team email here now

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml (hunk 2 above)
Comment (nickmcmahon01) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml]: I have made this change to our dev and preprop environments

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml (hunk 3 above)
Comment (nickmcmahon01) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/02-limitrange.yaml]: I have made this change to our dev and preprop environments

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml (hunk 4 above)
Comment (nickmcmahon01) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml]: done to both preprod and dev

File: namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml (hunk 5 above)
Comment (nickmcmahon01) [reply namespaces/live-1.cloud-platform.service.justice.gov.uk/sentence-planning-preprod/03-resourcequota.yaml]: changed this for dev and preprod

**Comment by nickmcmahon01** [comment]
I've made all the requested changes.
I've taken the name issue to the team - because we already have a dev environment once we have an answer we'll address both environments at the same time.

**Pull request merged**